src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const recursion = (num) => {
+    if (typeof num === 'number') {
+      sum += num;
+
+      return recursion;
+    } else {
+      const sumForReturn = sum;
+
+      sum = 0;
+
+      return sumForReturn;
+    }
```

Comment:
The current implementation of the `recursion` function checks the type of `num` which is unnecessary according to the task's description. You should remove the type check and directly add the number to the sum if it's provided, or return the current sum and reset it if no argument is passed.